### PR TITLE
Backend redux immutable integration

### PR DESCRIPTION
**Pull request by bethqiang**
Partially addresses #13 

Big changes made, other than the obvious addition of Redux (and mostly since the end of the working day today):
* Instead of emitting `position` and `rotation` _objects_ on each `tick`, the payload has been split up into its individual parts (x position, y position, z position, x rotation, y rotation, z rotation). This is to mirror our initial `User` constructor and the format that's consistent throughout the rest of our code; otherwise, our `merge` in the reducer gets a little wonky.
* Instead of having our initial users state be a `List`, I've converted it into a `Map`. This is so that instead of having to iterate through the `List` every time to find the specific user we're referring to (which happened relatively frequently), we can instead just refer to each user `Map` by its key (which, not coincidentally, is its id).
* Instead of the frontend using a `setInterval` to request updates from the backend, the backend is now pushing those updates to the frontend.

### CODE REVIEW

File: db/models/user.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const User = db.define('users', {
       notEmpty: true
     }
   },
-  // OAuth -> users may or may not have passwords.
+  // OAuth -> users may or may not hav e passwords.
```
Comment (parky22) [inline db/models/user.js]: weird space in 'have'

**Review by leafoflegend — COMMENTED** [review]
Line 31 of server/socket.js -> Why are we setIntervaling? Why are we not simply `subscribed` to the redux store on the back and having events based on the changes instead of firing 20 times a second? If the argument is throttling, you can set bounds on how often a change will affect emits.

Line 24-34 of browser/sockets/index.js seems like something you could potentially use a class or a function for. Just seems like something that could be a bit cleaner.

File: db/models/user.js (hunk 1 above)
Comment (bethqiang) [reply db/models/user.js]: Weird, I didn't touch the models at all...not sure how that happened haha.

**Comment by bethqiang** [comment]
@leafoflegend let's talk about both of these this afternoon! We talked about the second one as a team yesterday and couldn't really find a nice & practical way to make it work, but definitely down to hear other ideas.

**Review by JoeyDarbyshire — APPROVED** [review]

**Pull request merged**